version: '3.7'

services: 
  portainer:
    restart: unless-stopped
    image: portainer/portainer-ce:latest
    container_name: portainer
    volumes: 
      - /var/run/docker.sock:/var/run/docker.sock
      - /var/lib/docker/volumes:/var/lib/docker/volumes
      - portainerData:/data
    networks:
      - web
    ports: 
      - 127.0.0.1:9443:9443
      - 127.0.0.1:8000:8000
    labels: 
      - "traefik.enable=true"
      - "traefik.http.routers.portainer.entrypoints=web"
      - "traefik.http.routers.portainer.rule=Host(`portainer.${HOSTNAME}`)"
      - "traefik.http.routers.portainer-secure.rule=Host(`portainer.${HOSTNAME}`)"
      - "traefik.http.routers.portainer-secure.entrypoints=websecure"
      - "traefik.http.routers.portainer-secure.tls.certresolver=le"
      - "traefik.http.services.portainer.loadbalancer.server.port=9001"
    command: 
      --admin-password ${ADMIN_PASSWORD}

volumes: 
  portainerData:
    name: portainerData

networks:
  web:
    name: web
    external: true